ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -159,11 +157,21 @@ def add_experiences(
                 elif not curr_agent_step.done:
                     self.episode_steps[global_id] += 1
 
-            self.last_step_result[global_id] = batched_step_result
+            for _gid in action_global_agent_ids:
+                if _gid in self.last_step_result:
+                    if not self.last_step_result[_gid][0].done:
+                        if "action" in take_action_outputs:
+                            self.policy.save_previous_action(
+                                [global_id], take_action_outputs["action"]
+                            )
+                    else:
+                        # If it was done, delete it
+                        del self.last_step_result[_gid]
 
-        if "action" in take_action_outputs:
-            self.policy.save_previous_action(
-                previous_action.agent_ids, take_action_outputs["action"]
+            # Index is needed to grab from last_take_action_outputs
+            self.last_step_result[global_id] = (
```

Comment:
_gid ?

Reply:
This one needs to be global_id, its in a different loop